tests/integration/state_sim_tests/test_statevec.py

```diff
@@ -291,3 +318,59 @@ def test_all_gate_circ(simulator):
         # Measure
         qc.append({"Measure": {0, 1, 2, 3, 4}})
         check_measurement(simulator, qc)
+
+
+@pytest.mark.parametrize(
+    "simulator",
+    [
+        "MPS",
+    ],
+)
+def test_hybrid_engine_no_noise(simulator):
+    """Test that HybridEngine can use these simulators"""
+    check_dependencies(simulator)
+
+    n_shots = 1000
+    phir_folder = Path(__file__).parent.parent / "phir"
+
+    results = HybridEngine(qsim=simulator).run(
+        program=json.load(Path.open(phir_folder / "bell_qparallel.json")),
+        shots=n_shots,
+    )
+
+    m = results["m"]
+    assert np.isclose(m.count("00") / n_shots, m.count("11") / n_shots, atol=0.1)
+
+
+@pytest.mark.parametrize(
+    "simulator",
+    [
+        "MPS",
+    ],
+)
+def test_hybrid_engine_noisy(simulator):
+    """Test that HybridEngine with noise can use these simulators"""
+    check_dependencies(simulator)
+
+    n_shots = 1000
+    phir_folder = Path(__file__).parent.parent / "phir"
+
+    generic_errors = GenericErrorModel(
+        error_params={
+            "p1": 2e-1,
+            "p2": 2e-1,
+            "p_meas": 2e-1,
+            "p_init": 1e-1,
+            "p1_error_model": {
+                "X": 0.25,
+                "Y": 0.25,
+                "Z": 0.25,
+                "L": 0.25,
+            },
+        },
+    )
+    sim = HybridEngine(qsim=simulator, error_model=generic_errors)
+    sim.run(
+        program=json.load(Path.open(phir_folder / "example1_no_wasm.json")),
+        shots=n_shots,
+    )

```

Comment:
These two new tests take a long time to run (a total of 200s in my laptop for all MPS tests combined). The reason is that there are 2000 shots in total, and each shot takes about 200ms. I hope this is not an issue, since this tests are optional anyway (and will be skipped automatically if you don't have the cuQuantum and pytket-cutensornet requirements installed). Let me know if this is a problem, though, and I can write a less computationally intense test (although with fewer shots it would be hard to check for consistency of the distribution with theoretical probabilities).

Note: These circuits are too small for us to see benefit from using TN over statevector, and they suffer from major bottlenecks on the current Python implementation within pytket-cutensornet. Users should be discouraged from using TN methods on such small circuits.